doc/source/train/user_guide.rst

```diff
@@ -540,26 +540,61 @@ equivalent to the sum of the number of epochs.
     from ray import train
     from ray.train import Trainer
 
+    import torch
+    import torch.nn as nn
+    from torch.optim import Adam
+    import numpy as np
+
+
     def train_func(config):
-        model = 0 # This should be replaced with a real model.
-        for epoch in range(config["num_epochs"]):
-            model += epoch # This should be replaced with training logic.
-            train.save_checkpoint(epoch=epoch, model=model)
+        n = 100
+        # create a toy dataset
+        # data   : X - dim = (n, 4)
+        # target : Y - dim = (n, 1)
+        Y = torch.Tensor(np.random.uniform(0, 1, size=(n, 1)))
+        X = torch.Tensor(np.random.normal(0, 1, size=(n, 4)))
+
+        # toy neural network : 1-layer
+        model = nn.Linear(4, 1)
+        mse = nn.MSELoss()
+        optimizer = Adam(model.parameters(), lr=3e-4)
 
-    trainer = Trainer(backend="torch", num_workers=2)
+        for epoch in range(config["num_epochs"]):
+            # compute loss
+            y = model.forward(X)
+            loss = mse(y, Y)
+            # back-propagate loss
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            # For TensorFlow (Keras) models, use model.get_weights()
+            train.save_checkpoint(epoch=epoch, model_weights=model.state_dict())
+
+
+    trainer = Trainer(backend="torch", num_workers=1)
     trainer.start()
 
     trainer.run(train_func, config={"num_epochs": 1})
     print(trainer.latest_checkpoint)
-    # {'epoch': 0, 'model': 0}
+    # {'epoch': 0,
+    # 'model_weights': OrderedDict([('weight', tensor([[-0.4763, -0.3775,  0.4768,  0.3969]])),
+    #                               ('bias', tensor([0.2237]))]),
+    # '_timestamp': 1638979904}
 
     trainer.run(train_func, config={"num_epochs": 5})
     print(trainer.latest_checkpoint)
-    # {'epoch': 4, 'model': 10}
+    # {'epoch': 4,
+    # 'model_weights': OrderedDict([('weight', tensor([[ 0.4854, -0.1520, -0.1217,  0.2099]])),
+    #                               ('bias', tensor([-0.0724]))]),
+    # '_timestamp': 1638979904}
+
```

Comment:
We can remove these lines altogether, looking at it again now I don't think they add much value to the example.